main.js

```diff
@@ -0,0 +1,66 @@
+console.log('holla from js');
+
+var expenseName = document.forms[0].querySelector('input[name="expense-name"]');
+var expenseAmount = document.forms[0].querySelector('input[name="expense-amount"]');
+var expenseTableHead = document.querySelector('.expenses').children[0];
+var expenseTableBody = document.querySelector('.expenses').children[1];
+var inputForm = document.querySelector('.inputForm');
+var pressButton = document.querySelector('#add-expense');
+var totalExpense = document.querySelector('.expense-total');
+// var tableArr = [];
+
+var handleClick = function(evt) {
+  //gaurd against blank rows
+  if (expenseName.value === '' || expenseAmount.value === '') {
+    //broken error message next to button
+    // var error = document.createElement('p')
+    // error.textContent = 'Blank expenses, scrub?'
+    // inputForm.appendChild(error);
+    alert('Enter your expenses, scrub');
+  } else {
+  //create new table row
+    var newRow = document.createElement('tr');
+    newRow.innerHTML = '<td>' + expenseName.value + '</td><td>' + expenseAmount.value + '</td><button class="deleteButton" type="button">Delete</button></td>';
+    expenseTableBody.appendChild(newRow);
+    //push rows as objects to tableArr
+    tableArr = [];
+    for (var i = 0; i < expenseTableBody.rows.length; i++){
+      tableArr.push({
+        name: expenseTableBody.rows[i].cells[0].textContent,
+        amount: expenseTableBody.rows[i].cells[1].textContent
+      })
+    }
+    var sum = tableArr.reduce(function(a,b){
+      return +a + +b.amount;
+    }, 0);
+    // console.log(sum);
+    totalExpense.textContent = sum;
+    // expenseName.value = '';
+    // expenseAmount.value = '';
+  }
+
+};
+
+
+pressButton.addEventListener('click', handleClick);
+
+var buttonDelete = function(evt) {
+  if (event.target.classList.contains('deleteButton')){
+    event.target.parentNode.remove();
+    var newTotal = +totalExpense.textContent - event.target.previousSibling.textContent;
+    totalExpense.textContent = newTotal;
+  }
+}
+
+expenseTableBody.addEventListener('click', buttonDelete);
+
+//SORT
+
```

Comment:
You're right, it will be more efficient to keep the data separately in an array and sort that. Then update the representation of it in the DOM.